book/plugins/general-abilities/multiple-types-one-repo/multiple-types-one-repo.md

```diff
@@ -0,0 +1,37 @@
+# Using One Code Base for Multiple Plugins
+
+A developer can use one codebase for handling different plugin types.
+
+The main difference between [plugin types](/getting-started/plugin-types.md) and [plugin abilities](/getting-started/plugin-abilities.md) is the fact a single plugin can implement multiple abilities but can be of one type.
+
+For example, you can build a push plugin and later add support for an analytics plugin in the same codebase.
+
+The key idea for managing such a scenario is "separation of concerns". Each plugin type should be implemented on a different class.
+
+Then, you will need to create a different manifest file per plugin type implementation and publish it on Zapp.
+
+To do so please do the following steps:
+
+1. Create a separate manifest for each plugin type.
+2. Publish them to Zapp while making sure the dependency versions match between the manifests.
+
+The differences between these manifests would be:
+
+* Plugin type.
+* Plugin id must be different
```

Comment:
1. if you add `.` at the end (what you don't have to do since it's not really a sentence - just a bullet) - be consistent
2. Plugin ID , no need to add must be different - you already pointed out that these are the differences

Reply:
Dome